firmware/hw_layer/rtc_helper.cpp

```diff
@@ -83,97 +102,53 @@ static void put2(int offset, char *lcd_str, int value) {
 		lcd_str[offset + 1] = buff[1];
 	}
 }
+#endif // EFI_RTC
+
 
 /**
  * @return true if we seem to know current date, false if no valid RTC state
  */
 bool dateToStringShort(char *lcd_str) {
+#if EFI_RTC
 	strcpy(lcd_str, "000000_000000\0");
-	tm timp;
-	date_get_tm(&timp);
-	if (timp.tm_year < 116 || timp.tm_year > 130) {
+	efidatetime_t dateTime = getRtcDateTime();
+	if (dateTime.year < 2016 || dateTime.year > 2030) {
 		// 2016 to 2030 is the valid range
 		lcd_str[0] = 0;
 		return false;
 	}
 
-	put2(0, lcd_str, timp.tm_year % 100);	// Years since 1900 - format as just the last two digits
-	put2(2, lcd_str, timp.tm_mon + 1);		// months since January	0-11
-	put2(4, lcd_str, timp.tm_mday);			// day of the month	1-31
+	put2(0, lcd_str, dateTime.year % 100);		// year, format as just the last two digits
+	put2(2, lcd_str, dateTime.month);		    // month	1-12
+	put2(4, lcd_str, dateTime.day);				// day of the month	1-31
 
-	put2(7, lcd_str, timp.tm_hour);			// hours since midnight	0-23
-	put2(9, lcd_str, timp.tm_min);			// Minutes
-	put2(11, lcd_str, timp.tm_sec);			// seconds
+	put2(7, lcd_str, dateTime.hour);			// hours since midnight	0-23
+	put2(9, lcd_str, dateTime.minute);			// minutes
+	put2(11, lcd_str, dateTime.second);			// seconds
 
 	return true;
+#else // EFI_RTC
+	lcd_str[0] = 0;
+	return false;
+#endif // EFI_RTC
 }
 
 void dateToString(char *lcd_str) {
+#if EFI_RTC
 	// todo:
 	// re-implement this along the lines of 	chvprintf("%04u-%02u-%02u %02u:%02u:%02u\r\n", timp.tm_year + 1900, timp.tm_mon + 1, timp.tm_mday, timp.tm_hour,
 	// timp.tm_min, timp.tm_sec);
 	// this would require a temporary mem stream - see datalogging and other existing usages
 
 	strcpy(lcd_str, "00/00 00:00:00\0");
-	tm timp;
-	date_get_tm(&timp);			// get RTC date/time
+	efidatetime_t dateTime = getRtcDateTime();
 	
-	put2(0, lcd_str, timp.tm_mon + 1);
-	put2(3, lcd_str, timp.tm_mday);
-	put2(6, lcd_str, timp.tm_hour);
-	put2(9, lcd_str, timp.tm_min);
-	put2(12, lcd_str, timp.tm_sec);
-}
-
-static const char* const months[] = { "Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec" };
-
-void printDateTime() {
-	tm timp;
-
-	time_t unix_time = GetTimeUnixSec();
-	if (unix_time == -1) {
-		efiPrintf("invalid time in RTC cell");
-	} else {
-		efiPrintf("Current Unix time: %d", unix_time);
```

Comment:
Is this log still useful; should I add a conversion so we can log it?  not super excited to do so, re: leap seconds etc. ...